src/idlemonitor/DefaultIdleMonitor.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import PingPong from '../pingpong/PingPong';
+import PingPongObserver from '../pingpongobserver/PingPongObserver';
+import IdleMonitor from './IdleMonitor';
+
+export default class DefaultIdleMonitor implements IdleMonitor, PingPongObserver {
+  private canReceivePongs: boolean = false;
+  private lastPongResponseSeen: number = Date.now();
+
+  constructor(private pingPong: PingPong, private maxIdleTimeMs: number) {
+    this.addToPingPongObserver(this.pingPong);
+  }
+
+  addToPingPongObserver(pingPong: PingPong): void {
+    this.pingPong = pingPong;
+    this.pingPong.addObserver(this);
+  }
+
+  removeIdleMonitorObserver(): void {
+    this.canReceivePongs = false;
+    this.pingPong.removeObserver(this);
+  }
+
+  private timeSinceLastActiveMs(): number {
+    if (!this.canReceivePongs) {
+      return 0;
```

Comment:
Technically, returning `0` means that the session is active if this component cannot receive a pong. This does not make sense. Recommend making the following changes:

- Rename `this.canReceivePongs` to `this.enabled` (or other generic name).
- Set `this.enabled` to `true` in `start()`.
- Set `this.enabled` to `false` in `stop()`. `this.enabled` just tells whether `DefaultIdleMonitor` is enabled or not.
- In `isIdle()`, return `false` for the case you cannot determine whether it's idle or not.
  ```
   isIdle(): boolean {
    if (!this.enabled) {
      return false;
    }
    return this.timeSinceLastActiveMs() >= this.maxIdleTimeMs;
  }
  ```